test/system/030-run.bats

```diff
@@ -1465,4 +1465,28 @@ search               | $IMAGE           |
     is "$output" "Error.*: $expect" "podman emits useful diagnostic when no entrypoint is set"
 }
 
+@test "podman run - stopping loop" {
+    skip_if_remote "this doesn't work with with the REST service"
+
+    run_podman run -t -d --name testctr --stop-timeout 240 $IMAGE sh -c 'trap "echo READY" INT; while:; do sleep 0.1; done'
```

Comment:
Need a space between `while` and the colon.

You can run this yourself via
```console
$ ./bats --rootless 030:"stopping loop"
```
(I include --rootless simply to avoid the sudo password thing, because there's no difference here anyway)

Reply:
You also need to `s/INT/TERM/`, and possibly nuke the `-t`, but I *still* can't get this test to work.

Reply:
Added another wait_for_ready to make sure `sh -c` is good to go before we trigger the first stop - maybe that'll do it. Will check if it fixed things tomorrow.

Reply:
Oh it's much worse than that. I lied: you can't use `wait_for_ready` because that function "helpfully" checks container status and bails if != running. 

I'm working on something. Will let you know.

Reply:
This _passes_, but I suspect it's not quite exactly what you had hoped for. In particular, the `-t0` in the last `podman stop`. I can't make anything work without that.
```bash
@test "podman run - stopping loop" {
    skip_if_remote "this doesn't work with with the REST service"

    run_podman run -d --name testctr --stop-timeout 240 $IMAGE sh -c 'echo READY; sleep 999'
    cid="$output"
    wait_for_ready testctr

    run_podman inspect --format '{{ .State.ConmonPid }}' testctr
    conmon_pid=$output

    ${PODMAN} stop testctr &
    stop_pid=$!

    timeout=20
    while :;do
        sleep 0.5
        run_podman container inspect --format '{{.State.Status}}' testctr
        if [[ "$output" = "stopping" ]]; then
            break
        fi
        timeout=$((timeout - 1))
        if [[ $timeout == 0 ]]; then
            run_podman ps -a
            die "Timed out waiting for testctr to acknowledge signal"
        fi
    done

    kill -9 ${stop_pid}
    kill -9 ${conmon_pid}

    PODMAN_TIMEOUT=5 run_podman 125 stop -t0 testctr
    is "$output" "Error: container $cid conmon exited prematurely, exit code could not be retrieved: internal libpod error" "correct error on missing conmon"

    # This should be safe because stop is guaranteed to call cleanup?
    run_podman inspect --format "{{ .State.Status }}" testctr
    is "$output" "stopped" "container has successfully transitioned to exited state after stop"
}
```